### PR TITLE
Rodolfo/add viz component

### PR DESCRIPTION
**Pull request by RodolfoAllendes**
Basic display of networks fetched using the API, including different shape and colour for mRNA and microRNA elements.

User interaction is limited to selection of the layout used for the network display, and the display of the node ID (or label) at the bottom of the graph.

### CODE REVIEW

File: angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts]: Could this be private rather than protected? Alternatively I think it would be fine to just make it a bare public property without a getter.

File: angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts]: I think this could just be a simple public member without a getter/setter.

File: angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts:41]: I think this, along with colspan, could just be simple public properties without getters/setters.

File: angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: Is this `type` property going to be used in the future? Just wondering because I don't think it gets read anywhere.

File: angular/src/app/viewer/visualization/network-display/network-display.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/network-display/network-display.component.ts]: I think this is the `console.log` I put in, right? I think it would be fine to get rid of it (along with the `AfterViewInit` implementation), unless you want to keep it in a little longer for debugging purposes.

File: angular/src/app/viewer/visualization/network-display/network-display.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/network-display/network-display.component.ts:86]: Thanks for fixing the linter errors here! I think it might be possible to simplify this logic but I'll think about it and make a PR at some point.

File: angular/src/app/viewer/visualization/network-display/network-display.component.ts
Comment (yujiberra) [inline angular/src/app/viewer/visualization/network-display/network-display.component.ts]: Might be okay to delete these, unless you're planning on uncommenting in a future refactoring or something.

**Review by yujiberra — COMMENTED** [review]
@RodolfoAllendes 
Thanks for your work on this, and for taking care of the linter errors!

I tried this out locally and it seems to be working correctly, but I noticed one issue: if I generate a network A, and then change the gene set and then generate a new network B, what I actually end up with in the canvas is a new network with all the nodes and edges from both A and B. I think this is because the `set network` setter in `DisplayCanvasComponent` doesn't clear the old network.

I also noticed an issue where the "Generate network" button enabled state doesn't become disabled after switching tabs and coming back (which it should, because at that point there's no sample group or gene set selected), but that's a bug in my code; I'll work on a fix later.
(EDIT: I investigated this a little more and it'll be easy to fix but I'll wait until this PR gets merged because otherwise the merge will be a bit annoying)

I left a number of comments as well, they're almost all very minor points though.

File: angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts
Comment (RodolfoAllendes) [reply angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: its used as the return type for the changeLayout function.. 
basically `any` cant be used as the type anything (I think) if you want to avoid lint errors..

Comment (yujiberra) [reply angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: @RodolfoAllendes 
Isn't the return type of `changeLayout` void https://github.com/Toxygates/Panomicon/pull/19/files/95d601804075ad8a4b14a0a7736e4b6cdc9a59a1#diff-860375f887be92f6b7e531ec0394b882c2c3dd7e03e6fdd8a5fadda356f8aea2R47 ? Or maybe I am misunderstanding...

I did do a search for all references to the `type` property and didn't find any, but it could just be the editor + language system not being smart.

Also just to make sure, I'm not talking about the interface (`Layout`) is a whole, but the 'type: string' property, which I see is given a value whenever you declare instances of `Layout`, but which doesn't ever seem to be referred to.

Comment (RodolfoAllendes) [reply angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: your right xD.. the return is void... the parameter the function gets `options: Layout` is the one that uses the interface..

as for the type property, that one gets emited as an event and is captured by the display-canvas component (parent component) that uses it to actually update the layout of the visualization.. it is a bit hard to follow as the link is done in the display-canvas.html file rather on the ts  (I had to look through to find it now too XD)

Comment (yujiberra) [reply angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: @RodolfoAllendes 
Thanks for the clarification!

Sorry, can you do me a favor and link me to the location where the `type` property gets used? I just installed the Angular Language Service extension for VS Code so now "Find All References" does find references in HTML files as well, but I still wasn't able to find it :sweat_drops: 
(I also tried just searching for the string 'type' in the source and template files for LayoutPicker and DisplayCanvas but couldn't seem to find it...)

Comment (RodolfoAllendes) [reply angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: @yujiberra of course I've been looking at this wrong the whole time.. thinking of the `name` rather than the `type` property, even when you made it clear twice already that it was the latter...

I remember including these at some point because the configurable options for each layout are different and, when using typescript, by specifying the type of options will give you access to them... then, I noticed that, with the level of detail that I am currently implementing, I probably dont need them, and hence, no reference (you were right since the beginning).

If not too bothered, I would like to keep them a bit longer hanging around though, just in case I do end up using them in the future..

Comment (yujiberra) [reply angular/src/app/viewer/visualization/layout-picker/layout-picker.component.ts:47]: @RodolfoAllendes 
Thanks for the explanation, makes sense! 

> If not too bothered, I would like to keep them a bit longer hanging around though, just in case I do end up using them in the future..

Yup yup no problem at all!

File: angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts
Comment (yujiberra) [reply angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts:41]: @RodolfoAllendes 
Thanks for all the fixes! I think there might just be this one left:
It turns out I was lying here because the setter for `fetching` has a side effect, but I think `colspan` and `isEmpty` could be simple public properties?

Comment (RodolfoAllendes) [reply angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts:41]: @yujiberra 
I added the functionality on the setter for fetching on one of the latest commits...  so it was like that before

The colspan is meant to be similar in the future, specially when dealing with the theoretical display of multiple networks... but yeah, in the meantime it could simply be a public property

and the isEmpty is definetely ok being public

Comment (yujiberra) [reply angular/src/app/viewer/visualization/display-canvas/display-canvas.component.ts:41]: @RodolfoAllendes 
> The colspan is meant to be similar in the future, specially when dealing with the theoretical display of multiple networks... but yeah, in the meantime it could simply be a public property

Ah gotcha, in that case I have no problem with keeping colspan as is for now!

**Review by yujiberra — APPROVED** [review]
@RodolfoAllendes 
Thanks for the additional changes! I've confirmed that the network is cleared before fetching a new one.
I also see that both isEmpty and colspan are just regular properties now :pray:
LGTM!

(I imagine you're done for the week by now so I'll probably just merge this myself over the weekend)

**Pull request merged**
